File: Java/Tema3_Ejercicio9.java
/*
 * TEMA 3: ESTRUCTURAS CONDICIONALES - EJERCICIO 9
 * =================================================
 * Escriba un programa que determine el descuento a aplicar según el monto.
 */

import java.util.Scanner;

public class Tema3_Ejercicio9 {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        
        System.out.print("Digite el monto de la compra: $");
        double monto = scanner.nextDouble();
        
        System.out.println("\n=== CÁLCULO DE DESCUENTO ===");
        System.out.println("Monto original: $" + String.format("%.2f", monto));
        
        double descuento;
        String porcentajeDesc;
        
        if (monto < 50) {
            descuento = 0;
            porcentajeDesc = "0%";
        } else if (monto < 100) {
            descuento = monto * 0.05;
            porcentajeDesc = "5%";
        } else if (monto < 500) {
            descuento = monto * 0.10;
            porcentajeDesc = "10%";
        } else {
            descuento = monto * 0.20;
            porcentajeDesc = "20%";
        }
        
        double montoFinal = monto - descuento;
        
        System.out.println("Descuento: " + porcentajeDesc + " (-$" + String.format("%.2f", descuento) + ")");
        System.out.println("Monto final: $" + String.format("%.2f", montoFinal));
        
        scanner.close();
    }
}

/*
 * SALIDA ESPERADA:
 * Digite el monto de la compra: $250
 * 
 * === CÁLCULO DE DESCUENTO ===
 * Monto original: $250.00
 * Descuento: 10% (-$25.00)
 * Monto final: $225.00
 */

File: JS/Tema3_Ejercicio9.js
/*
 * TEMA 3: ESTRUCTURAS CONDICIONALES - EJERCICIO 9
 * ================================================
 * Calcular descuento según monto de compra.
 * 
 * EXPLICACIÓN:
 * - Descuentos en cascada según rango de cantidad
 * - Múltiples condicionales con rangos
 * - Cálculo de porcentaje
 */

function calcularDescuento(monto) {
    let porcentaje = 0;
    
    if (monto >= 1000) {
        porcentaje = 20;
    } else if (monto >= 500) {
        porcentaje = 15;
    } else if (monto >= 100) {
        porcentaje = 10;
    } else if (monto >= 50) {
        porcentaje = 5;
    }
    
    const descuento = monto * (porcentaje / 100);
    const montoFinal = monto - descuento;
    
    return {monto, porcentaje, descuento, montoFinal};
}

console.log("=== DESCUENTOS POR COMPRA ===");
const montos = [30, 75, 150, 500, 1000, 1500];

montos.forEach(monto => {
    const resultado = calcularDescuento(monto);
    console.log(`Compra de $${resultado.monto}: ${resultado.porcentaje}% descuento = $${resultado.montoFinal.toFixed(2)}`);
});

/*
 * SALIDA ESPERADA:
 * === DESCUENTOS POR COMPRA ===
 * Compra de $30: 0% descuento = $30.00
 * Compra de $75: 5% descuento = $71.25
 * Compra de $150: 10% descuento = $135.00
 * Compra de $500: 15% descuento = $425.00
 * Compra de $1000: 20% descuento = $800.00
 * Compra de $1500: 20% descuento = $1200.00
 */
